github/actions_secrets.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+type PublicKey struct {
+	ID  *string `json:"key_id,omitempty"`
+	Key *string `json:"key,omitempty"`
+}
+
+// GetPublicKey gets a public key that should be used for secret encryption.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/secrets/#get-your-public-key
+func (s *ActionsService) GetPublicKey(ctx context.Context, owner string, repo string) (*PublicKey, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/secrets/public-key", owner, repo)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pubKey := new(PublicKey)
+	resp, err := s.client.Do(ctx, req, pubKey)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pubKey, resp, nil
+}
+
+// Secret represents a repository action secret.
+type Secret struct {
+	Name      *string    `json:"name,omitempty"`
+	CreatedAt *Timestamp `json:"created_at,omitempty"`
+	UpdatedAt *Timestamp `json:"updated_at,omitempty"`
+}
+
+type secrets struct {
+	Secrets []*Secret `json:"secrets,omitempty"`
+}
+
+// ListSecrets lists all secrets available in a repository
+// without revealing their encrypted values.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/secrets/#list-secrets-for-a-repository
+func (s *ActionsService) ListSecrets(ctx context.Context, owner, repo string, opt *ListOptions) ([]*Secret, *Response, error) {
+	u := fmt.Sprintf("repos/%s/%s/actions/secrets", owner, repo)
+	u, err := addOptions(u, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	secrets := new(secrets)
+	resp, err := s.client.Do(ctx, req, &secrets)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return secrets.Secrets, resp, nil
+}
+
+// GetSecret gets a single secret without revealing its encrypted value.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/secrets/#get-a-secret
```

Comment:
Can you please contact support@github.com and let them know that they shouldn't need pagination for this endpoint, yet their documentation shows that they do?  (Or maybe just ask why they do... whichever you prefer.)
And then can you please report their answer back here?  They are usually incredibly fast in responding.
Thank you!

Reply:
I've reached out to GitHub support, more details below.